### PR TITLE
Monitoring CI

### PR DESCRIPTION
**Pull request by k7k0**

### CODE REVIEW

File: .github/workflows/monitoring-build-push-ecr.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        with:
+          access_key: ${{ env.AWS_ACCESS_KEY_ID }}
+          secret_access_key: ${{ env.AWS_SECRET_ACCESS_KEY }}
+          region: ${{ secrets.AWS_REGION }}
+
+      - name: Output Docker Image Tag
+        id: docker_meta
+        run: |
+          GITHEAD=`git rev-parse HEAD`
+          echo ::set-output name=tag::${GITHEAD}
```
Comment (archseer) [inline .github/workflows/monitoring-build-push-ecr.yml]: `set-output` is deprecated and will be removed by github next year. Modern github sets `GITHUB_REF_NAME` and `github.ref_name` so you can just use that

**Review by archseer — CHANGES_REQUESTED** [review]

File: .github/workflows/monitoring-build-push-ecr.yml (hunk 1 above)
Comment (k7k0) [reply .github/workflows/monitoring-build-push-ecr.yml]: done, thanks!

File: .github/workflows/monitoring-build-push-ecr.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      - name: Build & Publish Onchain Monitor to Registry
+        uses: elgohr/Publish-Docker-Github-Action@f7aca2fea76a5218f3c76cd5933c3ba1d8008774 # v3.02
+        with:
+          name: ${{ secrets.MONITORING_ECR_HOSTNAME }}
+          username: ${{ steps.ecr.outputs.username }}
+          password: ${{ steps.ecr.outputs.password }}
+          registry: ${{ steps.ecr.outputs.registry }}
+          tags: ${{ github.sha }}
+          context: monitoring
+          dockerfile: monitoring/ops/Dockerfile
```
Comment (archseer) [inline .github/workflows/monitoring-build-push-ecr.yml]: Could use the official docker actions as opposed to some third-party one: https://github.com/smartcontractkit/chainlink/blob/15d3f14cc8ddf264181e0b891c4b103fe71ea602/.github/actions/build-sign-publish-chainlink/action.yml#L88-L128

\cc @HenryNguyen5

File: .github/workflows/monitoring-build-push-ecr.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          region: ${{ secrets.AWS_REGION }}
+
+      - name: Build & Publish Onchain Monitor to Registry
+        uses: elgohr/Publish-Docker-Github-Action@f7aca2fea76a5218f3c76cd5933c3ba1d8008774 # v3.02
+        with:
+          name: ${{ secrets.MONITORING_ECR_HOSTNAME }}
+          username: ${{ steps.ecr.outputs.username }}
+          password: ${{ steps.ecr.outputs.password }}
+          registry: ${{ steps.ecr.outputs.registry }}
+          tags: ${{ github.sha }}
```
Comment (archseer) [inline .github/workflows/monitoring-build-push-ecr.yml]: not `github.ref_name`?

File: .github/workflows/monitoring-build-push-ecr.yml (hunk 2 above)
Comment (HenryNguyen5) [reply .github/workflows/monitoring-build-push-ecr.yml]: Agreed, this was from way before the official one existed! 

We are switching over to goreleaser soon though, may not be worth the extra effort

File: .github/workflows/monitoring-build-push-ecr.yml (hunk 3 above)
Comment (k7k0) [reply .github/workflows/monitoring-build-push-ecr.yml]: `github.ref_name` returns the current branch (or tag name), we use the commit sha as the image tag. I moved it to the `docker/metadata-action` approach, thanks!

File: .github/workflows/monitoring-build-push-ecr.yml (hunk 2 above)
Comment (k7k0) [reply .github/workflows/monitoring-build-push-ecr.yml]: @archseer done, thanks!
Also, I upgraded the versions of `docker/login-action` and `docker/metadata-action` of the linked example as they are using the deprecated methods as well 👍

**Review by archseer — APPROVED** [review]

**Pull request merged**